demoshell/main.py

```diff
@@ -69,6 +76,13 @@ def on_enter(self, key):
                 self._run_external_command(cmd)
             self.prompt_widget.set_edit_text('')
 
+        elif key == 'ctrl c':
+            if self.last_command and isinstance(
```

Comment:
I don't think we want ctrl-c to terminate the shell. That's not how most shells work, right? I think in the case where the last command has already terminated we can just ignore the ctrl-c or we could ring the bell (or flash the screen).

Reply:
Humm,ok, I got it. I think we can ignore and print it out like bellow.
```
daidv@DaiDV ~ $ ^C
daidv@DaiDV ~ $ ^C
daidv@DaiDV ~ $ ^C
```
To exit demoshell, just Ctl+D.

Is this Ok?

Reply:
![image](https://user-images.githubusercontent.com/10351187/50598986-21dae400-0edf-11e9-8f8f-154461acd65f.png)

I just try and rethink that it's better to ignore without print out anything.

Reply:
Hmm. A message wouldn't be bad, but ignoring the key entirely in the case where there is no active command would be OK as well.

Reply:
Ok, I will resolve the conflicts and merge this soon.